### PR TITLE
Adding BPT staking

### PR DESCRIPTION
**Pull request by weitianjie2000**

### CODE REVIEW

File: contracts/sNOTE.sol [hunk 1]
```diff
@@ -569,7 +575,7 @@ contract sNOTE is
 
         // NOTE: at this point the BPT has already been transferred into the sNOTE contract, so this
         // bptBalance amount includes bptAmount.
-        uint256 bptBalance = BALANCER_POOL_TOKEN.balanceOf(address(this));
+        uint256 bptBalance = LIQUIDITY_GAUGE.balanceOf(address(this));
```
Comment (jeffywu) [inline contracts/sNOTE.sol:578]: To be safe here, can we make an internal method that adds together the balances on both the POOL TOKEN and the LIQUIDITY GAUGE and then use that wherever we need to query the balanceOf

**Review by jeffywu — CHANGES_REQUESTED** [review]

File: scripts/environment.py [hunk 2]
```diff
@@ -216,12 +223,25 @@ def deployEmptyProxy(self):
         proxy = nProxy.deploy(emptyProxyImpl.address, bytes(), {"from": self.deployer})
         return Contract.from_abi("Proxy", proxy.address, EmptyProxy.abi)
 
-    def upgrade_sNOTE(self):
+    def deployBalancerToken(self):
+        return MockERC20.deploy("Mock Balancer Token", "BAL", 18, 0, {"from": self.deployer})
+
+    def upgrade_sNOTE(self, treasuryManager):
+        self.balancerMinter = MockBalancerMinter.deploy(self.bal, {"from": self.deployer})
```
Comment (jeffywu) [inline scripts/environment.py:230]: Instead of mocking these we should use the real ones:
    "LiquidityGauge": "0x40ac67ea5bd1215d99244651cc71a03468bce6c0",
    "BalancerMinter": "0x239e55F427D44C3cc793f49bFB507ebe76638a2b",
    
I think we should update the environment config to point to the real sNOTE pool and then just run these tests on a more recent block. It'll be more realistic and ensure we catch any other edge cases

**Review by jeffywu — APPROVED** [review]
